src/node.js

```diff
@@ -55,7 +55,10 @@ export let dedupeWalk = (fn, tree, { target = {}, dedupe } = {}) => {
   })(tree)
 }
 
-export let hasContext = node => node && node.context
+let isValue = x => !_.isNil(x) && (_.isNumber(x) || !_.isEmpty(x))
+
+export let hasContext = _.flow(_.get('context'), _.values, _.some(isValue))
```

Comment:
This fundamentally changes the meaning of `hasContext` - the idea was to be a quick heuristic to determine if a node supports returning results, not whether it has any. Additionally, reactors shouldn't care about the values of context results